app/helpers/discovered_hosts_helper.rb

```diff
@@ -66,4 +66,15 @@ def discovery_status_icon(host)
     "<span class='glyphicon #{status_glyph}' style='color: #{status_color}'
       title='#{status_message}'/>".html_safe
   end
+
+  def self.normalize_string_for_hostname(hostname)
+    hostname.to_s.downcase.gsub(/(^[^a-z0-9]|[^a-z0-9-_]|[^a-z0-9]$)/,'')
+  end
```

Comment:
Hosts must not start with a number, can you make sure this method checks and fix this?

Also this helper has zero unit test coverage, can you add new unit test and cover both methods?